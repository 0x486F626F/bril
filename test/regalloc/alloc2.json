{
  "functions": [
    {
      "instrs": [
        {
          "dest": "vj",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "dest": "vk",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "dest": "v1",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "dest": "v0",
          "op": "const",
          "type": "int",
          "value": 1
        },
        {
          "args": [
            "vj",
            "v1"
          ],
          "dest": "vg",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vk",
            "v0"
          ],
          "dest": "vh",
          "op": "sub",
          "type": "int"
        },
        {
          "args": [
            "vg",
            "vh"
          ],
          "dest": "vf",
          "op": "mul",
          "type": "int"
        },
        {
          "dest": "v2",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "args": [
            "vj",
            "v2"
          ],
          "dest": "ve",
          "op": "add",
          "type": "int"
        },
        {
          "dest": "v3",
          "op": "const",
          "type": "int",
          "value": 3
        },
        {
          "args": [
            "vj",
            "v3"
          ],
          "dest": "vm",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vf"
          ],
          "dest": "vb",
          "op": "id",
          "type": "int"
        },
        {
          "dest": "v4",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "args": [
            "ve",
            "v4"
          ],
          "dest": "vc",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vc"
          ],
          "dest": "vd",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "vm",
            "v4"
          ],
          "dest": "vk",
          "op": "add",
          "type": "int"
        },
        {
          "args": [
            "vb"
          ],
          "dest": "vj",
          "op": "id",
          "type": "int"
        },
        {
          "args": [
            "vj"
          ],
          "op": "print"
        },
        {
          "args": [
            "vk"
          ],
          "op": "print"
        },
        {
          "args": [
            "vd"
          ],
          "op": "print"
        }
      ],
      "name": "main"
    }
  ]
}
