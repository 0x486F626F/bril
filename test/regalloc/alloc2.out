main {
	r_04: int = const 1;
	vk: int = const 5;
	r_01: int = const 1;
	r_03: int = const 1;
	r_02: int = r_04 add r_01;
	v1: int = id r_01
	r_01: int = vk sub r_03;
	vf: int = id r_02;
	r_02: int = id vg;
	r_02: int = r_02 mul r_01;
	vh: int = id r_01
	r_01: int = const 2;
	r_03: int = r_04 add r_01;
	v2: int = id r_01
	r_01: int = const 3;
	vm: int = vj add v3;
	vj: int = id r_04
	r_04: int = id r_02;
	r_02: int = const 5;
	v3: int = id r_01
	r_01: int = r_03 add r_02;
	vc: int = id r_01
	vd: int = id r_01;
	r_01: int = id vc;
	r_01: int = id r_01;
	vk: int = vm add v4;
	vb: int = id r_04
	vj: int = id r_04;
	r_04: int = id vb;
	r_04: int = id r_04;
	print r_04;
	print vk;
	print r_01;
}
